src/compute.geometry/IO/Schema.cs

```diff
@@ -61,4 +67,215 @@ public ResthopperObject(object obj)
+        {
+            { GHTypeCodes.Boolean, "GH_Boolean" },
+            { GHTypeCodes.Point, "GH_Point" },
+            { GHTypeCodes.Vector , "GH_Vector" },
+            { GHTypeCodes.Integer , "GH_Integer" },
+            { GHTypeCodes.Number, "GH_Number" },
+            { GHTypeCodes.Text, "GH_String" },
+            { GHTypeCodes.Line, "GH_Line" },
+            { GHTypeCodes.Curve, "GH_Curve" },
+            { GHTypeCodes.Circle, "GH_Circle" },
+            { GHTypeCodes.PLane, "GH_Plane" },
+            { GHTypeCodes.Rectangle, "GH_Rectangle" },
+            { GHTypeCodes.Box, "GH_Box" },
+            { GHTypeCodes.Surface, "GH_Surface" },
+            { GHTypeCodes.Brep, "GH_Brep" },
+            { GHTypeCodes.Mesh, "GH_Mesh" }
+        };
+
+        public ResthopperInput()
+        {
+
+        }
+
+        public static T Cast<T>(object o)
+        {
+            return (T)o;
+        }
+
+        public bool TryBuildResthopperInput(string label, IGH_DocumentObject obj, out string message)
+        {
+            message = "";
+
+            // Attempt cast of generic object to grasshopper param
+            var param = obj as IGH_Param;
+            if (param == null)
+            {
+                message = "Invalid object in group.";
+                return false;
+            }
+
+            Param = param;
+
+            // Parse label for type information.
+            if (!TryParseName(label, out var msg))
+            {
+                message = msg;
+                return false;
+            }
+
+            return true;
+        }
+
+        public bool TrySetData(DataTree<ResthopperObject> tree)
+        {
+            Assembly assembly = Assembly.LoadFrom(@"C:\Program Files\Rhino WIP\Plug-ins\Grasshopper\Grasshopper.dll");
+            Assembly rhAssembly = Assembly.LoadFrom(@"C:\Program Files\Rhino WIP\System\RhinoCommon.dll");
+
+            Type paramType = assembly.GetType($"Grasshopper.Kernel.Parameters.{TypeCodeToParamType[TypeCode]}");
+
+            MethodInfo castMethod = this.GetType().GetMethod("Cast").MakeGenericMethod(paramType);
+            dynamic typedParam = castMethod.Invoke(null, new object[] { Param });
+
+            foreach (KeyValuePair<string, List<ResthopperObject>> entree in tree)
+            {
+                GH_Path path = new GH_Path(GhPath.FromString(entree.Key));
+
+                for (int i = 0; i < entree.Value.Count; i++)
+                {
+                    ResthopperObject restobj = entree.Value[i];
+                    var data = GetGeometry(restobj.Data, TypeCode);
+                    dynamic ghObj = Activator.CreateInstance(assembly.GetType($"Grasshopper.Kernel.Types.{TypeCodeToValueType[TypeCode]}"), data);
+                    typedParam.AddVolatileData(path, i, ghObj);
+                }
+            }
+
+            return true;
+        }
+
+        private dynamic GetGeometry(string json, GHTypeCodes type)
```

Comment:
haven't figured out dynamic deserialization just yet...but I think this is better than the repeating code from before